tests/rptest/tests/raft_availability_test.py

```diff
@@ -222,6 +222,22 @@ def test_one_node_down(self):
         the cluster remains available afterwards, and that the expected
         peer takes over as the new leader.
         """
+        # Retry a couple of times to establish new leader in a single election.
+        # This is racy if we just run it once. Running it multiple times
+        # gives us more confidence to actually detect a regression.
+        # Ref: https://github.com/vectorizedio/redpanda/issues/2444
+        retries = 4
+        ex: Exception = None
+        for i in range(1,retries+1):
+            try:
+                self._one_node_down()
+                return
+            except MetricCheckFailed as ex:
+                self.logger.info("Failed to elect new leader in single election, retrying "
+                    + "attempt ({i} of {retries}")
+        raise ex
```

Comment:
I don't think `ex` is ever set and remains `None`?

Reply:
you could remove that by counting the retries in the exception block and directly raising if you exceed the retry threshold. that'd also eliminate the type annotation.